tests/components/template/test_binary_sensor.py

```diff
@@ -523,11 +468,11 @@ async def test_invalid_attribute_template(hass, caplog):
     )
     await hass.async_block_till_done()
     assert len(hass.states.async_all()) == 2
-    await hass.helpers.entity_component.async_update_entity(
-        "binary_sensor.invalid_template"
-    )
+    await hass.async_start()
+    await hass.async_block_till_done()
 
-    assert ("Error rendering attribute test_attribute") in caplog.text
+    assert "test_attribute" in caplog.text
+    assert "TemplateError" in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)